examples/compute_polar_coordinates.py

```diff
@@ -0,0 +1,362 @@
+# the positive x-axis to the vector is anti-clockwise.
+
+# %%
+# Histogram of ``rho`` values
+# ----------------------------
+# Since ``rho`` is the distance between the ears' midpoint and the snout,
+# we would expect ``rho`` to be approximately constant in this data. We can
+# check this by plotting a histogram of its values across the whole clip.
+
+fig, ax = plt.subplots(1, 1)
+
+rho_data = head_vector_polar.sel(individuals=mouse_name, space_pol="rho")
+
+# compute and plot histogram
+ax.hist(
+    rho_data,
+    bins=50,
+)
+
+# add mean
+ax.axvline(
+    x=np.nanmean(rho_data),
+    c="b",
+    linestyle="--",
+)
+
+
+# add median
+ax.axvline(
+    x=np.nanmedian(rho_data),
+    c="r",
+    linestyle="-",
+)
+
+# add legend
+ax.legend(
+    [
+        f"mean = {np.nanmean(rho_data):.2f} pixels",
+        f"median = {np.nanmedian(rho_data):.2f} pixels",
+    ],
+    loc="best",
+)
+ax.set_ylabel("count")
+ax.set_xlabel(r"$\rho$" + " (pixels)")
+fig.show()
+
+# %%
+# We can see that there is some spread in the value of ``rho`` in this
+# dataset. This may be due to noise in the detection of the head keypoints,
+# or due to the mouse tipping its snout upwards during the recording.
+
+# %%
+# Histogram of ``phi`` values
+# -------------------------------------
+# We can also explore which ``phi`` values are most common in the dataset with
+# a circular histogram.
+
+# compute number of bins
+bin_width_deg = 5  # width of the bins in degrees
+n_bins = int(360 / bin_width_deg)
+
+# remove NaN values from the phi data
+bool_not_nan = (
+    ~np.isnan(
+        head_vector_polar.sel(individuals=mouse_name, space_pol="phi").values
+    ),
+)
+
+# compute histogram
+counts, bins = np.histogram(
+    head_vector_polar.sel(individuals=mouse_name, space_pol="phi").values[
+        bool_not_nan
+    ],
+    bins=np.linspace(-np.pi, np.pi, n_bins + 1),
+)
+
+
+# plot histogram as a bar plot in polar projection
+fig = plt.figure()
+ax = fig.add_subplot(projection="polar")
+
+bin_width_rad = np.deg2rad(bin_width_deg)
+bars = ax.bar(bins[:-1], counts, width=bin_width_rad, align="edge")
+
+ax.set_title("phi histogram")
+ax.set_theta_direction(-1)  # set direction clockwise
+ax.set_theta_offset(0)  # set zero at the right
+
+# set xticks to match the phi values in degrees
+n_xtick_edges = 9
+ax.set_xticks(np.linspace(0, 2 * np.pi, n_xtick_edges)[:-1])
+xticks_in_deg = (
+    list(range(0, 180 + 45, 45)) + list(range(0, -180, -45))[-1:0:-1]
+)
+ax.set_xticklabels([str(t) + "\N{DEGREE SIGN}" for t in xticks_in_deg])
+
+fig.show()
+
+# %%
+# The ``phi`` circular histogram shows that the head vector appears at a
+# variety of orientations in this dataset.
+
+# %%
+# Polar plot of the head vector within a time window
+# ---------------------------------------------------
+# We can also use a polar plot to represent the head vector in time,
+# in a coordinate system centred at the head of the mouse. Again,
```

Comment:
I was a bit thrown off by this phrasing. We are are still in coordinates relative to the frame (allocentric) here, right? As in, the phi angles ar relative to the frame's x-axis vector.

Reply:
Relatedly, when you say "from right to left" you mean "of the frame", right? Not necessarily "from the animal's right to the animal's left".

Reply:
Yeah, I think I tried to say it in simpler terms / less words and it was worse 😅 . 

In the following when I say "head centre" I mean "midpoint between the ears". I am also assuming the keypoints in the head of the mouse are a rigid body (that is, at constant distances).

> We are are still in coordinates relative to the frame (allocentric) here, right? As in, the phi angles ar relative to the frame's x-axis vector.

The head vector in this plot is represented in a coordinate system parallel to the pixel-space coordinate system but with it's origin at the mouse's head centre, at every frame. So as we go through frames, this "head coordinate system" is parallel to the pixel-space coord system, but its origin translates with the mouse's head centre. 

Since both coordinate systems are parallel, the theta angle computed relative to the x-axis in pixel-space is the same as the theta angle for the head coordinate system (because both x-axes are parallel). Does that make sense?

> Relatedly, when you say "from right to left" you mean "of the frame", right? Not necessarily "from the animal's right to the animal's left".

Well, in this case it would be the same, because both coordinate systems are always parallel. But I agree "left" and "right" of the animal/frame is ambiguous (depends on what is defined as forward right? I was trying to keep it simple sorry!).

I think the confusion comes from calling this "a coordinate system centred at the head of the mouse". While this is correct, it doesn't specify how this coordinate system is oriented at every frame. There are two coordinate systems that move with the mouse's head that are most used in this context:
- one would be a coordinate system whose origin is always at the mouse's head, but that is always parallel to the pixel-space coordinate system. Note that in this coordinate system, the snout of the mouse (or any other keypoint in the head) will change position as the mouse moves its head. This is what is represented in the plot (the head centre would be the origin, and the snout would be the markers).
- another coordinate system is one whose orientation is constant relative to the mouse's head. This coordinate system may be defined at every frame with its origin at the mouse's head centre, it's positive x-axis going from the head centre to the right ear, and it's y-axis perpendicular to the x-axis and positive going from the head centre to the snout. This coordinate system is "linked" to the head of the mouse at every frame: in this coordinate system the snout will be at a constant location throughout the clip (assuming the head is a rigid body). I think this is usually called 'egocentric' in neuroscience? I think you assumed this one right?

Will clarify in the text cell!

Reply:
Thanks for the clarification, I actually assumed the one you meant (the first option), it's just that the way it was phrased made me doubt it. I'm fine with the way it's worded now.
And correct, the 2nd coordinate system you refer to is the one referred to as "egocentric" (which is not what we have here).